src/main/java/org/opensearch/flowframework/transport/CreateWorkflowTransportAction.java

```diff
@@ -248,10 +249,19 @@ private void createExecute(WorkflowRequest request, User user, ActionListener<Wo
                                             ActionListener.wrap(stateResponse -> {
                                                 logger.info("Creating state workflow doc: {}", globalContextResponse.getId());
                                                 if (request.isProvision()) {
+                                                    // default to minus one indicate async execution
+                                                    TimeValue waitForTimeCompletion = TimeValue.MINUS_ONE;
```

Comment:
I see this block is repeated for both provision/reprovision cases. We can probably retrieve this once before any additional processing of the request